homotopy-core/src/contraction.rs

```diff
@@ -108,8 +108,9 @@ impl DiagramN {
                 signature,
             )?;
 
-            Ok(vec![cospan])
+            Ok::<_, ContractionError>(vec![cospan])
         })
+        .or(Err(ContractionError::Invalid))
```

Comment:
It feels very wrong to me to use `ContractionError::Invalid` here. This suggests the contraction was invalid (as in things didn't contract).